dll/win32/shell32/folders/CFSFolder.cpp

```diff
@@ -596,13 +596,34 @@ HRESULT SHELL32_GetFSItemAttributes(IShellFolder * psf, LPCITEMIDLIST pidl, LPDW
 
     BOOL bDirectory = (dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) != 0;
 
+    if (SFGAO_VALIDATE & *pdwAttributes)
+    {
+        STRRET strret;
+        if (SUCCEEDED(psf->GetDisplayNameOf(pidl, SHGDN_FORPARSING, &strret)))
+        {
+            LPWSTR path;
+            if (SUCCEEDED(StrRetToStrW(&strret, pidl, &path)))
+            {
+                BOOL exists = PathFileExistsW(path);
+                SHFree(path);
+                if (!exists)
+                    return E_FAIL;
+            }
+        }
+    }
+
     if (!bDirectory)
     {
         // https://git.reactos.org/?p=reactos.git;a=blob;f=dll/shellext/zipfldr/res/zipfldr.rgs;hb=032b5aacd233cd7b83ab6282aad638c161fdc400#l9
         WCHAR szFileName[MAX_PATH];
         LPWSTR pExtension;
+        BOOL hasName = _ILSimpleGetTextW(pidl, szFileName, _countof(szFileName));
 
-        if (_ILSimpleGetTextW(pidl, szFileName, _countof(szFileName)) && (pExtension = PathFindExtensionW(szFileName)))
+        // Hidden files with a leading tilde treated as super-hidden (devblogs.microsoft.com/oldnewthing/20170526-00/?p=96235#)
+        if (hasName && szFileName[0] == '~' && (dwFileAttributes & FILE_ATTRIBUTE_HIDDEN))
+            dwShellAttributes |= SFGAO_HIDDEN | SFGAO_SYSTEM;
```

Comment:
Does this happen also with folders, or just files? As far as I understand your change, you make it apply for both.

Reply:
This is inside `if (!bDirectory)`